src/frontend/src/utils/with_options.rs

```diff
@@ -226,18 +226,10 @@ impl TryFrom<&Statement> for WithOptions {
             Statement::CreateSource {
                 stmt:
                     CreateSourceStatement {
-                        with_properties,
-                        source_schema,
-                        ..
+                        with_properties, ..
                     },
                 ..
-            } => {
-                let mut options = with_properties.0.clone();
-                if let CompatibleSourceSchema::V2(source_schema) = source_schema {
-                    options.extend_from_slice(source_schema.row_options());
```

Comment:
Similar things are done for `create table` below. Did we forget to test that case?